pkgs/build-support/node/fetch-yarn-deps/index.js

```diff
@@ -154,9 +237,10 @@ const uniqueBy = (arr, callback) => {
 
 const prefetchYarnDeps = async (lockContents, verbose) => {
 	const lockData = lockfile.parse(lockContents)
+	const proxyAddr = getProxyUrl();
 	await performParallel(
 		uniqueBy(Object.entries(lockData.object), ([_, value]) => value.resolved)
-		.map(([key, value]) => () => downloadPkg({ key, ...value }, verbose))
+			.map(([key, value]) => () => downloadPkg({ key, ...value }, verbose, proxyAddr))
```

Comment:
In general, there are way to many functions changing signature. The proxy URL can be retrieved inside `downloadFileHttps`, which you forgot to fully rename.